holonote/annotate/plotting.py

```diff
@@ -0,0 +1,505 @@
+            return self._get_point_indices_by_position(**inputs)
+        else:
+            msg = f"{self.region_types} not implemented"
+            raise NotImplementedError(msg)
+
+    def register_tap_selector(self, element: hv.Element) -> hv.Element:
+        def tap_selector(x, y) -> None:  # Tap tool must be enabled on the element
+            # Only select the first
+            inputs = {str(k): v for k, v in zip(self.kdims, (x, y))}
+            indices = self.get_indices_by_position(**inputs)
+            if indices:
+                self.annotator.select_by_index(indices[0])
+            else:
+                self.annotator.select_by_index()
+
+        tap_stream = hv.streams.Tap(source=element, transient=True)
+        tap_stream.add_subscriber(tap_selector)
+        return element
+
+    def register_double_tap_clear(self, element: hv.Element) -> hv.Element:
+        def double_tap_clear(x, y):
+            self.clear_indicated_region()
+
+        double_tap_stream = hv.streams.DoubleTap(source=element, transient=True)
+        double_tap_stream.add_subscriber(double_tap_clear)
+        return element
+
+    def indicators(self) -> hv.DynamicMap:
+        self.register_tap_selector(self._element)
+        self.register_double_tap_clear(self._element)
+
+        def inner(_count):
+            return self.static_indicators
+
+        return hv.DynamicMap(inner, streams=[self._annotation_count_stream])
+
+    def overlay(self, indicators=True, editor=True) -> hv.Overlay:
+        layers = []
+        active_tools = []
+        if "range" in self.region_types or self.region_types == "point":
+            active_tools += ["box_select"]
+        elif self.region_types == "point-point":
+            active_tools += ["tap"]
+        layers.append(self._element.opts(tools=self.edit_tools, active_tools=active_tools))
+
+        if indicators:
+            layers.append(self.indicators())
+        if editor:
+            layers.append(self.region_editor())
+        return hv.Overlay(layers).collate()
+
+    @property
+    def static_indicators(self):
+        data = self.annotator.get_dataframe(dims=self.kdims)
+        fields_labels = self.annotator.all_fields
+        region_labels = [k for k in data.columns if k not in fields_labels]
+
+        indicator_kwargs = {
+            "data": data,
+            "region_labels": region_labels,
+            "fields_labels": fields_labels,
+            "invert_axes": False,  # Not yet handled
+            "groupby": self.annotator.groupby,
+        }
+
+        if self.region_types == "range":
+            indicator = Indicator.ranges_1d(**indicator_kwargs)
+        elif self.region_types == "range-range":
+            indicator = Indicator.ranges_2d(**indicator_kwargs)
+        elif self.region_types == "point":
+            indicator = Indicator.points_1d(**indicator_kwargs)
+        elif self.region_types == "point-point":
+            indicator = Indicator.points_2d(**indicator_kwargs)
```

Comment:
There should be a way to map to different versions of `Indicator.points_2d` i.e. crosshair versus a single point.